runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -542,7 +542,7 @@ ClientSessionData::getCHTable()
    {
    if (!_chTable)
       {
-      _chTable = new (PERSISTENT_NEW) JITServerPersistentCHTable(trPersistentMemory);
+      _chTable = new (PERSISTENT_NEW) JITServerPersistentCHTable(trPersistentMemory, jitConfig);
```

Comment:
Where does the jitConfig come from? Is this a global? If so, I would prefer to memorize in ClientSessionData. This would allow us in the future to have one jitConfig per client.

Reply:
> Where does the jitConfig come from? Is this a global? 

Yeah it's a global

> If so, I would prefer to memorize in ClientSessionData. This would allow us in the future to have one jitConfig per client.

Where would be the best place? Also Should it be the jitConfig from the client (i.e. remote pointer) or the jitConfig from the server?

Reply:
I am going back and forth on this one. Let's add `jitConfig` field in `ClientSessionData` and pass this field to `JITServerPersistentCHTable(trPersistentMemory, jitConfig);`
Initialization: ideally this field would be initialized in the constructor based on information from client. Unfortunately, when the object is constructed we don't know the value of client's jitconfig and we have to obtain it later through a message sent to client. Thus, this field needs a setter. I don't want to put the burden on you for implementing the required message exchange, so for the time being set this field based on jitConfig at the server. It should be available in `CompilationInfoPerThreadRemote` because that extends `CompilationInfoPerThreadBase` which has it. So, in `TR::CompilationInfoPerThreadRemote::processEntry()` you would do something like: `clientSession->setJitConfig(_jitConfig)`  (you may need a getter for jitConfig because I don't think it's accessible directly)